### PR TITLE
Template expansion order

### PR DESCRIPTION
**Pull request by tobyhede**
## Summary by CodeRabbit

* **New Features**
  * Runbook lists may use template variable references (e.g., {{ Target }}); defined refs resolve to concrete runbook paths, unresolved refs remain literal, and loop-scoped refs are preserved until iteration time.

* **Documentation**
  * Format/spec updated to describe template-backed runbook list entries and their resolution semantics.

* **Tests**
  * Added coverage for parsing, mixed literal/template lists, resolution behavior (CLI and renderer), FOR-scope rules, and edge cases.

### CODE REVIEW

File: packages/cli/src/services/template-renderer.ts [hunk 1]
```diff
@@ -410,23 +413,147 @@ function validatePromptedForSteps(steps: readonly ResolvedStep[]): void {
...
+/**
+ * Check whether a RunbookRef variable path is scoped to a FOR loop variable.
+ *
+ * Returns true when the ref is the loop variable itself (`item`) or a
+ * dotted path rooted on it (`item.runbook`). These refs are only resolvable
+ * at iteration time, so the resolver should preserve them as placeholder text.
+ */
+function isForScoped(ref: string, forVariable: string | undefined): boolean {
+  if (!forVariable) return false;
+  return ref === forVariable || ref.startsWith(forVariable + '.');
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/cli/src/services/template-renderer.ts]: Fixed in commit 05b96a72 — added JSDoc @param/@returns, template literal, and optional chain.

File: packages/parser/src/guards.ts [hunk 2]
```diff
@@ -166,6 +169,26 @@ export function isBoundRef(bound: Bound): bound is BoundRef {
...
 
+/**
+ * Type guard: checks if a runbook entry is an unresolved template reference.
+ *
+ * @param entry - The runbook entry to check
+ * @returns True if `entry` is a `RunbookRef` (`entry is RunbookRef`), guaranteeing `entry.ref` is a string
+ */
+export function isRunbookRef(entry: RunbookEntry): entry is RunbookRef {
+  return typeof entry === 'object' && 'ref' in entry;
+}
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/parser/src/guards.ts:180]: Not a real issue — `RunbookEntry = string | RunbookRef` excludes `null` at the type level. All call sites pass values from typed arrays validated by Zod schemas. No runtime path produces `null` here. The existing `isBoundRef` guard follows the same pattern.
